src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -52,11 +52,16 @@ internal override Cci.PrimitiveTypeCode GetPrimitiveTypeCode(PEModuleSymbol modu
             return type.PrimitiveTypeCode;
         }
 
-        internal override bool IsVolatileModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
+        internal override bool IsAcceptedVolatileModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
         {
             return type.SpecialType == SpecialType.System_Runtime_CompilerServices_IsVolatile;
         }
 
+        internal override bool IsAcceptedIsConstModifierType(TypeSymbol type)
+        {
+            return type.IsWellKnownTypeIsConst();
+        }
```

Comment:
Could we simplify `IsWellKnownTypeIsConst` by caching the `IsConst` type on the `PEModuleSymbol`, similar to the handling of `System.Type` in `SymbolFactory.GetSystemTypeSymbol`.